flink/src/main/java/org/apache/iceberg/flink/IcebergTableSource.java

```diff
@@ -122,6 +134,21 @@ public boolean isLimitPushedDown() {
 
   @Override
   public TableSource<RowData> applyLimit(long newLimit) {
-    return new IcebergTableSource(loader, schema, properties, projectedFields, true, newLimit);
+    return new IcebergTableSource(loader, schema, properties, projectedFields, true, newLimit, filters);
+  }
+
+  @Override
+  public TableSource<RowData> applyPredicate(List<Expression> predicates) {
+    List<org.apache.iceberg.expressions.Expression> expressions = Lists.newArrayList();
+    for (Expression predicate : predicates) {
+      FlinkFilters.convert(predicate).ifPresent(expressions::add);
+    }
+
+    return new IcebergTableSource(loader, schema, properties, projectedFields, isLimitPushDown, limit, expressions);
```

Comment:
@openinx, is this guaranteed to only be called on a source that has not had predicates pushed? This ignores the existing predicates in this source. Maybe we should add a precondition to check that assumption.

I think it is safe either way because this doesn't remove any predicates from the list. From reading the Javadoc, I think that will result in all predicates running in Flink also.